src/passport-saml/inmemory-cache-provider.ts

```diff
@@ -48,15 +48,15 @@ export class CacheProvider {
           nowMs >=
           new Date(this.cacheKeys[key].createdAt).getTime() + this.options.keyExpirationPeriodMs
         ) {
-          this.remove(key, () => undefined);
+          this.removeAsync(key);
```

Comment:
I understand that it is customary to use the `Async` suffix for Promise-returning functions, however, that seems to be an unnecessary breaking change as it would completely change the API for cache providers, which could break a lot of plugged-in code. Could we not rename the function, or would it be advisable to not append `Async` since we aren't actually making this newly async as it always was that way, just with callbacks?

Reply:
If I don't change method names, the problem will be much larger - cache functions will be called, but callback - will not. Instead of crashing on compile-time people will get a timeout in their production code